src/rngs/adapter/reseeding.rs

```diff
@@ -220,6 +237,40 @@ impl<R, Rsdr> CryptoRng for ReseedingCore<R, Rsdr>
 where R: BlockRngCore + SeedableRng + CryptoRng,
       Rsdr: RngCore + CryptoRng {}
 
+
+// Fork protection
+//
+// We implement fork protection on Unix using `pthread_atfork`.
+// When the process is forked, we increment `RESEEDING_RNG_FORK_COUNTER`.
+// Every `ReseedingRng` stores the last known value of the static in
+// `fork_counter`. If the cached `fork_counter` is less than
+// `RESEEDING_RNG_FORK_COUNTER`, it is time to reseed this RNG.
+//
+// If reseeding fails, we don't deal with this by setting a delay, but just
+// don't update `fork_counter`, so a reseed is attempted a soon as possible.
+#[cfg(all(feature="std", unix))]
+static mut RESEEDING_RNG_FORK_COUNTER: u64 = 0;
+
+#[cfg(all(feature="std", unix))]
+extern fn forkhandler() {
+    unsafe { RESEEDING_RNG_FORK_COUNTER += 1; }
+}
+
+#[cfg(all(feature="std", unix))]
+fn register_fork_handler() {
+    unsafe {
+        libc::pthread_atfork(None, None, Some(forkhandler));
```

Comment:
... to register a parameterless function updating a static variable. I don't think this approach is *wrong*, but it will probably update the counter more than necessary as soon as multiple threads/reseeding RNGs are used.

So use `std::sync::Once` to register `forkhandler` maybe?